relayer/lavasession/consumer_session_manager.go

```diff
@@ -75,11 +76,34 @@ func (csm *ConsumerSessionManager) atomicReadCurrentEpoch() (epoch uint64) {
 	return atomic.LoadUint64(&csm.currentEpoch)
 }
 
+// validating we still have providers, otherwise reset valid addresses list
+func (csm *ConsumerSessionManager) validatePairingListNotEmpty() uint64 {
+	csm.lock.RLock() // lock read to validate length
```

Comment:
this function layout is prune for errors, need to do:
bigFunc(){
should_reset := readLockedFunc(){
Rlock
defer Runlock
}
if should_reset {
WritelockedFunc(){
lock 
defer unlock
check again
do it
}